src/create-airseeker.ts

```diff
@@ -0,0 +1,187 @@
+import { Choice, PromptObject } from 'prompts';
+import { AirnodeRrpAddresses } from '@api3/airnode-protocol';
+import { Api, Beacon, OperationsRepository } from './types';
+import { promptQuestions } from './utils/prompts';
+import { readOperationsRepository } from './utils/read-operations';
+import { writeOperationsRepository } from './utils/write-operations';
+import { runAndHandleErrors } from './utils/cli';
+import { chainNameToChainId } from './utils/evm';
+import { Beacons, Gateways, Templates, Triggers } from './utils/airseeker-validation';
+import { sanitiseFilename } from './utils/filesystem';
+
+const questions = (choices: Choice[]): PromptObject[] => {
+  return [
+    {
+      type: 'text',
+      name: 'name',
+      message: ['What is the Airseeker configuration name?'].join('\n'),
+      initial: 'global',
+    },
+    {
+      type: 'autocompleteMultiselect',
+      name: 'selectedBeacons',
+      message: 'What are the beacons you want to include in the configuration?',
+      choices: choices,
+    },
+    //TODO: Add BeaconSet question
+  ];
+};
+
+const main = async () => {
+  const operationsRepository = readOperationsRepository();
+  const documentation = operationsRepository.documentation;
+  const beaconChoices = Object.values(operationsRepository.apis).flatMap((api) =>
+    Object.values(api.beacons).map((beacon) => ({
+      title: `${api.apiMetadata.name}\t${beacon.name}`,
+      value: { api: api, beacon: beacon },
+      selected: true,
+    }))
+  );
+
+  const response = await promptQuestions(questions(beaconChoices));
+  const selectedBeacons = response.selectedBeacons as { api: Api; beacon: Beacon }[];
+  const beacons = selectedBeacons.map((beacon) => beacon.beacon);
+  const apis = selectedBeacons
+    .map((beacon) => beacon.api)
+    .filter((api, index, apis) => apis.findIndex((find) => find.apiMetadata.name === api.apiMetadata.name) === index);
+
+  // Get all the chains the API will be deployed on
+  const apiChains = [...new Set(beacons.flatMap((beacon) => beacon.chains.map((chain) => chain.name)))];
+
+  //// Build airseeker.json ////
+  const airseekerBeacons = beacons.reduce(
+    (beaconObj, beacon) => ({
+      ...beaconObj,
+      [beacon.beaconId]: {
+        airnode: beacon.airnodeAddress,
+        templateId: beacon.templateId,
+        fetchInterval: beacon.airseekerConfig.updateInterval,
+      },
+    }),
+    {} as Beacons
```

Comment:
I know you have mentioned before about not inferring the type like this but this is a special case I encountered while using `reduce()`. Without using this `airseekerBeacons` shows its type as `{}` which is wrong but if it would still pass the check in `airseekerConfigSchema` because an empty object passes those checks. If there is something else that I might be doing wrong I'm all ears.